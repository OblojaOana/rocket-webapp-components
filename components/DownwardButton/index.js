import DownwardButton from "../../core/DownwardButton";
export default DownwardButton;