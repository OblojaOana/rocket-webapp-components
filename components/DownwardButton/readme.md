##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import DownwardButton from '@bit/totalsoft_oss.react-mui.downward-button';

export default (
	<DownwardButton
		title="Downward"
	></DownwardButton>
)
 ```