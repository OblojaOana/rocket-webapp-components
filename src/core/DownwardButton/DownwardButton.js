import React from "react";
import PropTypes from "prop-types";
import { ArrowDownward } from '@material-ui/icons';
import CustomIconButton from '../IconButton';

function DownwardButton({ title, onClick, disabled, fontSize, ...rest }) {
    return (
        <CustomIconButton aria-label="Downward" disabled={disabled} onClick={onClick} tooltip={title} {...rest}>
            <ArrowDownward fontSize={fontSize} />
        </CustomIconButton>
    )
}

DownwardButton.defaultProps = {
    color: "themeNoBackground",
    fontSize: "small",
    size: "medium"
};

DownwardButton.propTypes = {
    title: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground"
    ]),
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    fontSize: PropTypes.oneOf([
        "inherit",
        "default",
        "small",
        "large"
    ])
}

export default DownwardButton;