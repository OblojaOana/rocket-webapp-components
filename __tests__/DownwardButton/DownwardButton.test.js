import React from 'react';
import renderer from 'react-test-renderer';
import DownwardButton from '../../components/DownwardButton/DownwardButton';

describe('DownwardButton', () => {
    it('snapshot', () => {
        const component = renderer.create(<DownwardButton
            title="Downward"
            onClick={() => { }}
        ></DownwardButton>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});