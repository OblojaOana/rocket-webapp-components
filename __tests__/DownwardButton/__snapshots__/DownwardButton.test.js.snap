// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownwardButton snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Downward"
  className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17 Mui-disabled Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  title="Downward"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
      />
    </svg>
  </span>
</button>
`;
